### ISSUE TITLE
Traction Control Improvement

### ISSUE THREAD

**Issue by whitespace-rebel** [RigsOfRods/rigs-of-rods#3188]
Hi everyone,

I'm developing a high torque (660 Nm) RWD sports car, and it is really hard to drive with the keyboard even with traction control on.

The wheels push really hard especially in 3rd and 4th gear.
I looked into the code and realized that wheel slip is fixed at 25%, which is too high for this car.

In source/main/physics/BeamForcesEuler.cpp
you can see this: wheel_slip > 0.25f
It is used once for anti-lock braking and once for the traction control.

My suggestion is to make that 0.25 a variable.
In the .truck file we have the "TractionControl" settings where you can define the wheel slip, but currently it is unused.

Thank you!

**Comment by ohlidalp** [comment]
Hi,

[The docs](https://docs.rigsofrods.org/vehicle-creation/fileformat-truck/#tractioncontrol) link to specific commit which removed those, and it's by a programmer who was objectively good at tweaking the simulation for stability and reliability. I can't restore that slip parameter without potentially breaking older mods which have it but players rely on it being ignored. At the same time, there's multiple things that need research and tweaking (i.e. [clutch calculation](https://github.com/RigsOfRods/rigs-of-rods/pull/3177)) and I'm not good with math and algorithms, so I need to figure out some flexible diagnostic/tweaking system.

TL;DR I will give you some means of changing that value, but not via truck format, that's too rigid, it will be via scripting.

**Comment by whitespace-rebel** [comment]
This is realy nice, I like this solution.

I also thought about this problem. The only solution I could come up with is a version parameter inside the .truck file.
So if VERSION >= 12345 then read the .truck file according to this version.

Will this patch land in the dev build? https://rigs-of-rods.itch.io/rigs-of-rods-dev
BTW the hash on that site doesn't show any of the commits in the master branch. Is it up to date?
